pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/BaseFilterOperator.java

```diff
@@ -31,15 +34,29 @@
 
   private Predicate predicate;
   private int nextBlockCallCounter = 0;
+  private PredicateEvaluator _predicateEvaluator;
 
   public void setPredicate(Predicate predicate) {
     this.predicate = predicate;
   }
 
+  public void setPredicateEvaluator(PredicateEvaluator predicateEvaluator) {
```

Comment:
Predicate was being set from outside already. I am setting the evaluator from outside just as an optimization. The FilterPlanNode.constructPhysicalOperator does create the evaluator to optimize for alwaysFalse. It then calls this api to set what it created, just to avoid re-creation.